docusaurus/docs/create-a-plugin/extend-a-plugin/instrument-a-backend-plugin.md

```diff
@@ -0,0 +1,421 @@
+- **uname:** If available, the username of the user that made the request.
+
+**Example:**
+
+```go
+package plugin
+
+import (
+    "context"
+    "errors"
+
+    "github.com/grafana/grafana-plugin-sdk-go/backend"
+    "github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+)
+
+func NewDatasource(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+    // instantiates a logger that will include the datasource UID
+    // key/value pair over the life-time of this instance.
+    logger := backend.Logger.With("dsUID", settings.UID)
+
+    return &Datasource{
+        logger:   logger,
+    }, nil
+}
+
+func (ds *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+    ctxLogger := ds.logger.FromContext(ctx)
+    ctxLogger.Debug("Debug msg", "someID", 1)
+    ctxLogger.Info("Info msg", "queryType", "default")
+    ctxLogger.Warning("Warning msg", "key", "value")
+    ctxLogger.Error("Error msg", "error", errors.New("An error occurred"))
+}
+```
+
+#### Include additional contextual information in logs
+
+If you want to propagate additional contextual key/value pairs to subsequent code/logic you can use the [log.WithContextualAttributes](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend/log#WithContextualAttributes) function.
+
+**Example:**
+
+```go
+func (ds *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+    ctxLogger := ds.logger.FromContext(ctx)
+    ctxLogger.Debug("Debug msg", "someID", 1)
+    ctxLogger.Info("Info msg", "queryType", "default")
+    ctxLogger.Warning("Warning msg", "key", "value")
+    ctxLogger.Error("Error msg", "error", errors.New("An error occurred"))
+
+    for _, q := range req.Queries {
+        childCtx = log.WithContextualAttributes(ctx, []any{"refID", q.RefID, "queryType", q.QueryType})
+        ds.handleQuery(childCtx, q)
+    }
+}
+```
+
+### Best practices/recommendations
+
+- Start the log message with a capital letter, e.g. `logger.Info("Hello world")` instead of `logger.Info("hello world")`.
+- The log message should be an identifier for the log entry, try to avoid parameterization, e.g. `logger.Debug(fmt.Sprintf(“Something happened, got argument %d”, “arg”)), in favor of key-value pairs for additional data, e.g. `logger.Info(“Something happened”, “argument”, “arg”).
+- Prefer using camelCase style when naming log keys, e.g. remoteAddr or userID, to be consistent with Go identifiers.
+- Use the key `error` when logging Go errors, e.g. `logger.Error("Something failed", "error", errors.New("An error occurred")`.
+- Use a contextual logger whenever you have access to a `context.Context`.
+- Do not log sensitive information, such as data source credentials or IP addresses, or other personally identifiable information.
+
+#### Validate and sanitize input coming from user input
```

Comment:
Room for improvement/possible to skip this. I think the SDK/Grafana should validate/sanitize to ease developer experience.